docs/resources/extract-ai.sh

```diff
@@ -6,26 +6,84 @@ BINARY_FOLDER=/var/mnt
 
 pushd $FOLDER
 
-total_copies=$(sort -u $BINARY_FOLDER/$OCP_RELEASE_LIST | wc -l)  # Required to keep track of the pull task vs total
-current_copy=1
-
-while read -r line;
-do
-  uri=$(echo "$line" | awk '{print$1}')
-  #tar=$(echo "$line" | awk '{print$2}')
-  podman image exists $uri
-  if [[ $? -eq 0 ]]; then
-      echo "Skipping existing image $tar"
-      echo "Copying ${uri} [${current_copy}/${total_copies}]"
+load_images() {
+
+  declare -A pids # Hash that include the images pulled along with their pids to be monitored by wait command
+  local max_bg=10 # Max number of simultaneous skopeo copies to container storage
+  local total_copies=$(sort -u $BINARY_FOLDER/$OCP_RELEASE_LIST | wc -l)  # Required to keep track of the pull task vs total
+  local current_copy=1
+
+  #remove duplicates
+  sort -u -o $OCP_RELEASE_LIST $OCP_RELEASE_LIST
+
+  while read -r line;
+  do
+    uri=$(echo "$line" | awk '{print$1}')
+    podman image exists $uri
+    if [[ $? -eq 0 ]]; then
+      echo "[INFO] Skipping existing image $tar"
+      echo "[INFO] Copying ${uri} [${current_copy}/${total_copies}]"
       current_copy=$((current_copy + 1))
       continue
+    fi
+    tar=$(basename ${uri/:/_})
+    tar --use-compress-program=pigz -xf ${tar}.tgz
+    if [ $? -eq 0 ]; then rm -f ${tar}.gz; fi
```

Comment:
If we fail to extract the tarball, we're continuing on to try to skopeo copy whatever got extracted. Should we instead be reporting a failure and continuing to the next image?